docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,669 @@
+corresponds to. The operator will then issue a `GET` prior to attempting to update the secret to ensure that it owns the secret. Unfortunately, as far as I can
+tell KeyVault secrets don't support `Etag` so there's a possible data race here that we can't avoid...
+
+Conclusion: TBD
+
+**What happens when a resource is updated to remove the `secrets` entry?**
+
+If a resource is created and given a `forOperator.secrets` field that instructs it to create a secret, and then is updated to remove this secrets entry,
+should we delete the secret we previously created? It feels like we _should_, but when the operator is presented with the updated object during that second
+reconcile it doesn't know that it ever had a secret.
+
+Some possible solutions:
+
+1. We just don't delete the secrets when you do this and expect users to do it themselves if they want things cleaned up.
+2. For k8s secrets, we could probably look through all of the secrets in the resources namespace and see if any of them are owned by us. If there are some
+   and we don't have any `forOperator.secrets` we delete them. The downside here is that's a lot of overhead to do for all reconciles. This also assumes that
+   the operator has Secret `Read` permissions in the namespaces in question - which it's possible that it doesn't (it may never have created any secrets and
+   the users may know that it won't and so have denied it permission). This also obviously doesn't work for KeyVault secrets as we have no idea which KeyVault
+   the secrets might even be in.
+3. When we create a secret we store information about it in `Status` or `Annotations` and then use that information to clean up after ourselves. It's a bit 
+   cleaner to store it in `Status` but technically that can be lost, whereas `Annotations` won't be. This should work for both KeyVault and k8s secrets and 
+   while it's a bit icky I think gives the best experience.
+
+Conclusion: TBD
+
+**Should we drop the `Destination` suffix in the field names?**
+
+For example, the proposal for CosmosDB `DatabaseAccount` was:
+```yaml
+spec:
+  # Other spec fields elided...
+  forOperator:
+    secrets:
+      keyDestination:
+        # stuff elided...
+      readOnlyKeyDestination:
+        # stuff elided...
+```
+
+```yaml
+spec:
+  # Other spec fields elided...
+  forOperator:
+    secrets:
+      key:
+        # stuff elided...
+      readOnlyKey:
+        # stuff elided...
+```
+
+Conclusion: TBD
+
+**Do we allow writing a secret to a namespace where the resource isn't?**
+
+No, see: https://github.com/kubernetes/community/pull/5455
+
+Conclusion: TBD
+
+**Do we support writing multiple destinations at the same time?**
+
+If/when we add support for writing secrets to KeyVault rather than to Kubernetes secrets, would we also enable support for writing the same secret to _both_
+KeyVault and Kubernetes at the same time?
```

Comment:
At the same time in this context meant concurrently. If one or the other (or both) fails, we would return a reconciler error and attempt to re-reconcile (the same as if creating a resource in Azure failed, we keep trying and keep driving towards goal state).

Reply:
I've removed the "at the same time" language here, just because really my question is more about "do we support writing to both". Have also added an example `spec` highlighting what I am talking about here.